src/main/java/wordcloud/WordCloud.java

```diff
@@ -103,12 +104,18 @@ public WordCloud(int width, int height, CollisionMode collisionMode) {
 
     public void build(List<WordFrequency> wordFrequencies) {
         Collections.sort(wordFrequencies);
-
+        int curword = 1;
+        final Dimension dimensions = new Dimension(width, height);
         for(final Word word : buildwords(wordFrequencies, this.colorPalette)) {
-            final int startX = RANDOM.nextInt(Math.max(width - word.getWidth(), width));
-            final int startY = RANDOM.nextInt(Math.max(height - word.getHeight(), height));
-            place(word, startX, startY);
-
+            final Point p = startscheme.getStartingPoint(dimensions, word);
+            boolean placed = place(word, p.x, p.y);
```

Comment:
i prefer final everywhere i can (for variables).

Sorry I don't have a code guideline posted on the readme :)